memory/table.go

```diff
@@ -150,47 +150,35 @@ func NewPartitionedTableWithCollation(db *BaseDatabase, name string, schema sql.
 		}
 	}
 
+	stripTblNamesFunc := func(e sql.Expression) (sql.Expression, transform.TreeIdentity, error) {
```

Comment:
Good refactor, but might as well make this its own top-level function in this file instead of a local var